models/flow/d3q27_cumulant/Dynamics.R

```diff
@@ -25,44 +26,61 @@ AddSetting(name="GalileanCorrection",default=1.,comment='Galilean correction ter
 AddSetting(name="ForceX",default=0, comment='Force force X')
 AddSetting(name="ForceY",default=0,comment='Force force Y')
 AddSetting(name="ForceZ",default=0,comment='Force force Z')
+AddSetting(name="Smag", default=0, comment='Smagorinsky coefficient for SGS modeling')
+AddSetting(name="Omega", default=1, comment='relaxation rate for 3rd order cumulants')
 
 AddGlobal(name="Flux", comment='Volume flux', unit="m3/s")
+AddGlobal(name="Drag", comment='Force exerted on body in X-direction', unit="N")
+AddGlobal(name="Lift", comment='Force exerted on body in Z-direction', unit="N")
+AddGlobal(name="Lateral", comment='Force exerted on body in Y-direction', unit="N")
 
 AddNodeType("WVelocityTurbulent","BOUNDARY")
-AddNodeType("NSymmetry","BOUNDARY")
-AddNodeType("SSymmetry","BOUNDARY")
 AddNodeType("NVelocity", "BOUNDARY")
 AddNodeType("SVelocity", "BOUNDARY")
 AddNodeType("NPressure", "BOUNDARY")
 AddNodeType("SPressure", "BOUNDARY")
+AddNodeType("Buffer", "BOUNDARY")
+
+AddNodeType("NSymmetry", "ADDITIONALS")
+AddNodeType("SSymmetry", "ADDITIONALS")
+AddNodeType("Body", "BODY")
 	
 #Adding terms for supporting time-correlation for synthetic turbulence
 
 AddDensity( name="SynthTX",dx=0,dy=0,dz=0)
 AddDensity( name="SynthTY",dx=0,dy=0,dz=0)
 AddDensity( name="SynthTZ",dx=0,dy=0,dz=0)
 
+AddNodeType("IB",group="HO_BOUNDARY")
+for (f in fname) AddField(f,dx=0,dy=0,dz=0) # Make f accessible also in present node (not only streamed)
+
 #Averaging values
 if (Ave) {
```

Comment:
It would be good to make the averaging also an option.

Reply:
I've tried to do that. Unfortunately, encountered some problems along the way 
* Using #ifdef OPT_AVG here resulted in compilation error (similar for all averaged variables): 
Dynamics.c:71:16: error: ‘varUXUY’ was not declared in this scope
         st.z = varUXUY/average_iter - getavgU().x*getavgU().y;
                ^~~~~~~
makefile:48: recipe for target 'cuda.o' failed

* Additionally, I've tried to pass the value of OPT_AVG to Dynamics.R, not to create unnecessary fields if averaging is turned off, but it is not possible.